### PR TITLE
feat(core): `Tooltip` remove delay on click

### LINKED ISSUE #3450: 🐞 - tui-tooltip and tui-hint don't work if you quickly click multiple times on host element
### Which @taiga-ui/* package(s) are the source of the bug?

kit

### Please provide a link to a minimal reproduction of the bug

https://taiga-ui.dev/components/tooltip#example-base

### Is this issue blocking you?

Non-Blocking

### Description

Sometimes users instead of hover trying to click on elements which have tooltips. Currently if you try to click multiple times on host element it doesn't show tooltip (or close existing).

### Angular version

14.1.3

### Taiga UI version

3.15

### Which browsers have you used?

- [X] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3450 

## What is the new behavior?

We need to separate click$ stream from toggle$, because toggle() also called on hover from [HintComponent](https://github.com/Tinkoff/taiga-ui/blob/main/projects/core/directives/hint/hint.component.ts#L75)

We need to add zero delay to click$ stream, because without delay event bubbles up and [closes hint immediately](https://github.com/Tinkoff/taiga-ui/blob/main/projects/core/directives/hint/hint.component.ts#L86)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/core/directives/hint/hint-hover.directive.ts [hunk 1]
```diff
@@ -34,6 +38,10 @@ export class TuiHintHoverDirective extends TuiDriver {
     }
 
     @HostListener('click', ['true'])
+    onClick(visible: boolean): void {
```
Comment (waterplea) [inline projects/core/directives/hint/hint-hover.directive.ts:41]: Unnecessary argument

**Review by splincode — APPROVED** [review]

**Review by splincode — APPROVED** [review]

**Comment by waterplea** [comment]
Looks a bit like overcomplication to me :(

**Comment by vladimirpotekhin** [comment]
> Looks a bit like overcomplication to me :(

check other solution plz

**Comment by waterplea** [comment]
> > Looks a bit like overcomplication to me :(
> 
> check other solution plz

Now it's a bit hacky :( Could you check the following idea:

Do we need a delay on toggling to true at all? What if we remove HostListener on click and a showDelay for toggle$ and only trigger it from `Tooltip` component?

**Comment by vladimirpotekhin** [comment]
> > > Looks a bit like overcomplication to me :(
> > 
> > 
> > check other solution plz
> 
> Now it's a bit hacky :( Could you check the following idea:
> 
> Do we need a delay on toggling to true at all? What if we remove HostListener on click and a showDelay for toggle$ and only trigger it from `Tooltip` component?

I checked it. Toggle triggered from [hint](https://github.com/Tinkoff/taiga-ui/blob/main/projects/core/directives/hint/hint.component.ts#L75) and yes we need delay for it. I didn't find a solution without separating click into another stream (

**Comment by waterplea** [comment]
We don't really need delay for `true` values though. In hint we only care about delay for `false` so it's not immediately closed when we hover away from the hint bubble. What if we remove delay for `true` values in toggle?

**Review by waterplea — APPROVED** [review]

**Pull request merged**